test_vector_search_in_cloud.py

```diff
@@ -0,0 +1,274 @@
+DATASET_FILE_PATH = DATA_DIR_PATH + DATASET_FILENAME
+SCHEMA_FILE_PATH = DATA_DIR_PATH + SCHEMA_FILENAME
+
+ANN_SEARCH_CMD_TEMPLATE = "SELECT id FROM vdb_bench.vdb_bench_collection ORDER BY vector ANN OF {vector} LIMIT 10;"
+
+VECTOR_SEARCH_INSTANCE_TYPES = {
+    "aws": {"t4g.small": 175, "t4g.medium": 176, "r7g.medium": 177},
+    "gce": {"e2-small": 178, "e2-medium": 179, "n4-highmem-2": 180}
+}
+
+
+class VectorSearchInCloudBase(ClusterTester):
+    average_recall: float = 0.0
+
+    def get_email_data(self):
+        self.log.info("Prepare data for email")
+
+        email_data = self._get_common_email_data()
+        email_data.update({"average_recall": self.average_recall})
+
+        return email_data
+
+    @staticmethod
+    def download_vector_search_test_data_from_s3():
+        s3_storage = S3Storage(bucket=BUCKET_NAME)
+
+        base_url = f"https://{BUCKET_NAME}.s3.amazonaws.com"
+        s3_storage.download_file(link=f"{base_url}/{TEST_DATA_FILENAME}", dst_dir=DATA_DIR_PATH)
+        s3_storage.download_file(link=f"{base_url}/{GROUND_TRUTH_FILENAME}", dst_dir=DATA_DIR_PATH)
+
+        s3_storage.download_file(link=f"{base_url}/{SCHEMA_FILENAME}", dst_dir=DATA_DIR_PATH)
+        s3_storage.download_file(link=f"{base_url}/{DATASET_FILENAME}", dst_dir=DATA_DIR_PATH)
+
+    @cached_property
+    def test_data(self):
+        return File(TEST_DATA_FILE_PATH).readlines()
+
+    @cached_property
+    def ground_truth_data(self):
+        return File(GROUND_TRUTH_FILE_PATH).readlines()
+
+    def prepare_vector_search_index(self):
+        self.log.info("Download Vector Search test data from S3")
+        self.download_vector_search_test_data_from_s3()
+
+        self.log.info("Create schema applying CQL statements")
+        ks_statements, other_statements = get_schema_create_statements_from_file(schema_file_path=SCHEMA_FILE_PATH)
+
+        # DC name written in the schema file may differ from the one used in the cluster under test
+        # if run in region or cloud provider different from ones where backup was taken
+        dc_under_test_name = next(iter(self.db_cluster.get_nodetool_status()))
+        ks_statements = substitute_dc_name_in_ks_statements_if_different(ks_statements=ks_statements,
+                                                                         current_dc_name=dc_under_test_name)
+
+        with self.db_cluster.cql_connection_patient(node=self.db_cluster.nodes[0], verbose=False) as session:
+            for cql_stmt in ks_statements + other_statements:
+                session.execute(cql_stmt)
+
+            self.log.info("Insert sample data into the table")
+            insert_cmd_template = "INSERT INTO vdb_bench.vdb_bench_collection (id, vector) VALUES ({id}, {vector});"
+            with open(DATASET_FILE_PATH, encoding="utf-8") as dataset_file:
+                for line in dataset_file:
+                    id_str, vector_str = line.strip().split(",", maxsplit=1)
```

Comment:
there better way to read CSV files..

also we should consider using cqlsh for such a thing. (as we instruct costumers)

Reply:
https://docs.scylladb.com/manual/stable/cql/cqlsh.html#copy-from